app/helpers/application_helper.rb

```diff
@@ -82,6 +82,14 @@ def cloudinary(url, width = "500", quality = 80, format = "auto")
     Images::Optimizer.call(SimpleIDN.to_ascii(image_url), width: width, quality: quality, fetch_format: format)
   end
 
+  def optimized_image_tag(image_url, optimizer_options: {}, image_options: {})
+    image_options[:width] ||= optimizer_options[:width]
```

Comment:
Whats the point of having `width` and `height` in the  optimizer_options and then transferring them to the image_options? Why not set them in the image_options to begin with?

This also seems pretty similar to the method above in terms of its function. Any reason not to combine them so we have a single method handling `Images::Optimizer.call` in all the views? Then if we update the Optimizer we only have to change it in one place.

Reply:
Width and Height will be used by both the optimizer (`cl_image_path`) and the `image_tag`. Technically the value will be used by the optimizer first that's why I did it this way. I can see the scenario where we provide a different W/H for `image_option` and/or not provide W/H to `optimizer_options` (see note below). Perhaps I can change `image_options` to `image_tag_options` so it's more clear?

The method above, `#cloudinary`, is quite similar to barebone `Images::Optimizer` and I was putting it aside when I was consolidating `cl_image_tag. I can do that refactor in a different PR. I know it's already in the codebase, but Is it a bit weird to be calling service object from the view?

misc note: `cl_image_tag` will only take width and height into account when `crop` isn't nil but these values will always be applied to the image tag's property. We don't have `crop: nil` anywhere so that doesn't affect us.

Reply:
> I can see the scenario where we provide a different W/H for image_option and/or not provide W/H to optimizer_options

Gotcha, thanks for the explanation! 

> but Is it a bit weird to be calling service object from the view?

I wouldn't call the optimizer service from the view but I would consolidate the helper methods that call it into one single helper method so its all consolidated. But like you mentioned a good refactor PR once you iron out all the functionality you need.

Reply:
> This also seems pretty similar to the method above in terms of its function. Any reason not to combine them 

When I suggested this helper in [this comment](https://github.com/forem/forem/pull/9834#discussion_r471910795), I only used the context of this PR, so didn't notice there's a related helper around. Sometimes you just get stuck in a local maximum 😉 But 👍 for combining them in a refactor PR.